File: services/ms-cliente/java/airportsystem/mscliente/dto/EnderecoDTO.java
package airportsystem.mscliente.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

public class EnderecoDTO {

    @NotBlank(message = "CEP é obrigatório")
    @Size(min = 8, max = 8, message = "CEP deve ter 8 dígitos")
    private String cep;

    @NotBlank(message = "UF é obrigatório")
    private String uf;

    @NotBlank(message = "Cidade é obrigatório")
    private String cidade;

    @NotBlank(message = "Bairro é obrigatório")
    private String bairro;

    @NotBlank(message = "Rua é obrigatório")
    private String rua;

    @NotBlank(message = "Número é obrigatório")
    private String numero;
    
    private String complemento;

    public String getCep() {
        return cep;
    }

    public void setCep(String cep) {
        this.cep = cep;
    }

    public String getUf() {
        return uf;
    }

    public void setUf(String uf) {
        this.uf = uf;
    }

    public String getCidade() {
        return cidade;
    }

    public void setCidade(String cidade) {
        this.cidade = cidade;
    }

    public String getBairro() {
        return bairro;
    }

    public void setBairro(String bairro) {
        this.bairro = bairro;
    }

    public String getRua() {
        return rua;
    }

    public void setRua(String rua) {
        this.rua = rua;
    }

    public String getNumero() {
        return numero;
    }

    public void setNumero(String numero) {
        this.numero = numero;
    }

    public String getComplemento() {
        return complemento;
    }

    public void setComplemento(String complemento) {
        this.complemento = complemento;
    }

}

File: services/ms-cliente/airport-system.ms-cliente/src/main/kotlin/com/tads/airport_system/ms_cliente/dto/EnderecoDTO.kt
package com.tads.airport_system.ms_cliente.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

data class EnderecoDTO(
    @field:NotBlank(message = "CEP é obrigatório")
    @field:Size(min = 8, max = 8, message = "CEP deve ter 8 dígitos")
    val cep: String,

    @field:NotBlank(message = "UF é obrigatória")
    val uf: String,

    @field:NotBlank(message = "Cidade é obrigatória")
    val cidade: String,

    @field:NotBlank(message = "Bairro é obrigatório")
    val bairro: String,

    @field:NotBlank(message = "Rua é obrigatória")
    val rua: String,

    @field:NotBlank(message = "Número é obrigatório")
    val numero: String,

    val complemento: String?
) 